sql/information_schema/information_schema.go

```diff
@@ -514,11 +517,102 @@ func collationsRowIter(ctx *Context, c *Catalog) RowIter {
 			c.PadSpace(),
 		})
 	}
-	return RowsToRowIter(rows...)
+	return RowsToRowIter(rows...), nil
 }
 
-func emptyRowIter(ctx *Context, c *Catalog) RowIter {
-	return RowsToRowIter()
+func triggersRowIter(ctx *Context, c *Catalog) (RowIter, error) {
+	var rows []Row
+	for _, db := range c.AllDatabases() {
+		triggerDb, ok := db.(TriggerDatabase)
+		if ok {
+			triggers, err := triggerDb.GetTriggers(ctx)
+			if err != nil {
+				return nil, err
+			}
+			var triggerPlans []*plan.CreateTrigger
+			for _, trigger := range triggers {
+				parsedTrigger, err := parse.Parse(ctx, trigger.CreateStatement)
+				if err != nil {
+					return nil, err
+				}
+				triggerPlan, ok := parsedTrigger.(*plan.CreateTrigger)
+				if !ok {
+					return nil, ErrTriggerCreateStatementInvalid.New(trigger.CreateStatement)
+				}
+				triggerPlans = append(triggerPlans, triggerPlan)
+			}
+
+			triggerPlans = analyzer.OrderTriggers(triggerPlans)
+			var beforeDelete []*plan.CreateTrigger
+			var beforeInsert []*plan.CreateTrigger
+			var beforeUpdate []*plan.CreateTrigger
+			var afterDelete []*plan.CreateTrigger
+			var afterInsert []*plan.CreateTrigger
+			var afterUpdate []*plan.CreateTrigger
+			for _, triggerPlan := range triggerPlans {
+				switch triggerPlan.TriggerTime {
+				case sqlparser.BeforeStr:
+					switch triggerPlan.TriggerEvent {
+					case sqlparser.DeleteStr:
+						beforeDelete = append(beforeDelete, triggerPlan)
+					case sqlparser.InsertStr:
+						beforeInsert = append(beforeInsert, triggerPlan)
+					case sqlparser.UpdateStr:
+						beforeUpdate = append(beforeUpdate, triggerPlan)
+					}
+				case sqlparser.AfterStr:
+					// OrderTriggers reverses AFTER, so we must reverse them again
+					switch triggerPlan.TriggerEvent {
+					case sqlparser.DeleteStr:
+						afterDelete = append([]*plan.CreateTrigger{triggerPlan}, afterDelete...)
+					case sqlparser.InsertStr:
+						afterInsert = append([]*plan.CreateTrigger{triggerPlan}, afterInsert...)
+					case sqlparser.UpdateStr:
+						afterUpdate = append([]*plan.CreateTrigger{triggerPlan}, afterUpdate...)
+					}
+				}
+			}
+
+			for _, planGroup := range [][]*plan.CreateTrigger{beforeDelete, beforeInsert, beforeUpdate, afterDelete, afterInsert, afterUpdate} {
```

Comment:
Is this order part of a spec, or just arbitrary? Should add documentation for how you determined this

Reply:
Also if it is part of a spec, should be a test verifying the behavior.

Reply:
Order and grouping are completely arbitrary. Each group (BEFORE/AFTER + INSERT/DELETE/UPDATE combination) has its own order of execution, so this is just an efficient way for me to loop through everything and use the index (since they're sorted by execution order) as the execution order. I'll add a comment nonetheless.